From da205253f0d78204900ec82b6cb2cb0afd9f1254 Mon Sep 17 00:00:00 2001
From: Stanislav Levin <slev@altlinux.org>
Date: Thu, 8 Aug 2019 19:43:12 +0300
Subject: [PATCH] Fix Pytest 5 errors

This fixes
"""
pytest.PytestDeprecationWarning: raises(..., 'code(as_a_string)')
is deprecated, use the context manager form or use `exec()` directly.
"""

Fixes: https://github.com/RonnyPfannschmidt/iniconfig/issues/6
Signed-off-by: Stanislav Levin <slev@altlinux.org>
---
 test_iniconfig.py | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/test_iniconfig.py b/test_iniconfig.py
index 27fd53f..fe12421 100644
--- a/test_iniconfig.py
+++ b/test_iniconfig.py
@@ -150,27 +150,25 @@ def test_iniconfig_from_file(tmpdir):
     assert list(config.sections) == ['metadata']
     config = IniConfig(path, "[diff]")
     assert list(config.sections) == ['diff']
-    py.test.raises(TypeError, "IniConfig(data=path.read())")
+    with pytest.raises(TypeError):
+        IniConfig(data=path.read())
 
 
 def test_iniconfig_section_first(tmpdir):
-    excinfo = py.test.raises(ParseError, """
+    with pytest.raises(ParseError) as excinfo:
         IniConfig("x", data='name=1')
-    """)
     assert excinfo.value.msg == "no section header defined"
 
 
 def test_iniconig_section_duplicate_fails():
-    excinfo = py.test.raises(ParseError, r"""
+    with pytest.raises(ParseError) as excinfo:
         IniConfig("x", data='[section]\n[section]')
-    """)
     assert 'duplicate section' in str(excinfo.value)
 
 
 def test_iniconfig_duplicate_key_fails():
-    excinfo = py.test.raises(ParseError, r"""
+    with pytest.raises(ParseError) as excinfo:
         IniConfig("x", data='[section]\nname = Alice\nname = bob')
-    """)
 
     assert 'duplicate name' in str(excinfo.value)
 
-- 
2.28.0

